site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java

```diff
@@ -13182,159 +13126,46 @@ private void setSelectedParticipantRoles(SessionState state) {
 
 	} // setSelectedParticipantRol3es
 
+	@Getter @AllArgsConstructor
 	public class MyIcon {
-		protected String m_name = null;
-
-		protected String m_url = null;
-
-		protected String m_skin = null;
-
-		public MyIcon(String name, String url, String skin) {
-			m_name = name;
-			m_url = url;
-			m_skin = skin;
-		}
-
-		public String getName() {
-			return m_name;
-		}
-
-		public String getUrl() {
-			return m_url;
-		}
-
-		public String getSkin() {
-			return m_skin;
-		}
+		private String name = null;
+		private String url = null;
+		private String skin = null;
 	}
 
 	// a utility class for working with ToolConfigurations and ToolRegistrations
 	// %%% convert featureList from IdAndText to Tool so getFeatures item.id =
 	// chosen-feature.id is a direct mapping of data
+	@Getter @EqualsAndHashCode
```

Comment:
getOuterType seems a little confusing as its not really referring to Class hierarchy but rather its just comparing the instance of SiteAction. Which then comes Adrian's last question about is there multiple instances of SiteAction? I don't see any code calling **new** but there are multiple servlets declared in the web.xml specifically one for each tool sakai.sitemanage, sakai.sitebrowser, sakai.siteinfo, sakai.sitesetup.
So the answer is yes there can be multiple instances of SiteAction. Its possible that lombok has this covered as with this line in their equals `if (!super.equals(o)) return false;` so we maybe fine with lombok.